### PR TITLE
Add the git-submodules-status script

### PR DESCRIPTION
**Pull request by gkotian**
The script fetches all submodules of a project from their respective
remote repositories, and shows a table that lists for each submodule :
the latest available version, the current version and a small visual cue
to easily identify mismatches between the two.

### CODE REVIEW

File: git-submodules-status [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#
+#   Description:
+#   ------------
+#       Fetches all submodules from their respective remote repositories, and
+#       shows a table that lists for each submodule : the latest available
+#       version, the current version and a small visual cue to easily identify
+#       mismatches between the two.
+#
+#   Note that this script assumes that a project's submodules are present in a
+#   directory called "submodules" in the project's top-level git directory.
```
Comment (leandro-lucarella-sociomantic) [inline git-submodules-status]: I don't think this is true anymore, it seems like the information about the submodules is being retrieved from `.gitmodules`.

Comment (leandro-lucarella-sociomantic) [inline git-submodules-status]: Or maybe I'm wrong.

File: git-submodules-status [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    echo "The following submodules haven't been initialized : ${UNINITIALISED_SUBMODS[@]}"
+    echo "Try running this first:"
+    echo "    git submodule update --init"
+    exit 2
+fi
+
+# Read the existing submodules information into an array. After this, the array
+# should contain an even number of entries, with each pair of entries containing
+# the submodule name and the current submodule version respectively.
+ARR=(`git submodule | sed 's|^.*submodules/\(.*\) (\(.*\))$|\1 \2|g'`)
```
Comment (leandro-lucarella-sociomantic) [inline git-submodules-status]: Ha, this is the line that makes it depends on being in `submodules/`? But this is not really needed to be part of the regex, right?

**Comment by leandro-lucarella-sociomantic** [comment]
Not super important at this stage, but if the restriction about having to put submodules in the `submodules/` directory can be removed, it could be a better start.

**Comment by gkotian** [comment]
> Not super important at this stage, but if the restriction about having to put submodules in the submodules/ directory can be removed, it could be a better start.

True. I'll try to get rid of this restriction.

File: git-submodules-status [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+THIRD_COLUMN_WIDTH=12
+
+# Write the header row and its underlines
+echo "$(appendSpaces "Submodule" $FIRST_COLUMN_WIDTH) Current version     Latest version"
+echo "---------     ---------------     --------------"
+
+# Write each submodule's data
+for ((i = 0; i < ${#ARR[@]}; i+=2))
+do
+    cd $GIT_TOP/submodules/${ARR[$i]}
```
Comment (gkotian) [inline git-submodules-status]: This line is another reason for the `submodules` directory restriction. But even this can be dealt with I think.

Comment (llucax) [reply git-submodules-status]: Yes, getting the info from .gitmodules should be fairly easy.

**Comment by gkotian** [comment]
Pushed an updated version without the `submodules/` directory restriction.

File: git-submodules-status [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+fi
+
+# Read the existing submodules information into an array. After this, the array
+# should contain an even number of entries, with each pair of entries containing
+# the path of the submodule (relative to the top-level git directory) and the
+# current submodule version (within parentheses) respectively.
+SUBMODS_INFO_ARR=(`git submodule status | sed 's/^\s\+//g' | cut -d" " -f2-`)
+
+# Fetch each submodule from the remote repo so that we can get its latest tag
+git submodule foreach git remote update -p
```
Comment (leandro-lucarella-sociomantic) [inline git-submodules-status]: Why the `-p`? It looks weird to have a "destructive" operation here?

**Comment by leandro-lucarella-sociomantic** [comment]
Other than that last question, LGTM.

File: git-submodules-status (hunk 4 above)
Comment (gkotian) [reply git-submodules-status]: No reason, apart from it being my personal preference when I update my repos. But you're right; there's no reason for that to be in this script. I'll remove it.

**Comment by gkotian** [comment]
Updated with the `-p` option removed in the `git remote update` command.

(also sorry for the delay... was on vacation, and only now beginning to catch up with emails)

**Pull request merged**